pyccel/stdlib/lists/lists.c

```diff
@@ -0,0 +1,665 @@
+        list->size -= 1;
+        return (ret_val);
+    }
+
+    return (NULL);
+}
+
+void    free_pop(t_pop_ret** pop_val)
+/*
+        Deallocate the t_pop_ret object resulting from a pop operation
+
+        Parameters
+        ----------
+        pop_val : t_pop_ret**
+            double pointer on the t_pop_ret object to be deallocated
+*/
+{
+    if (pop_val && (*pop_val))
+    {
+        free((*pop_val)->raw);
+        free(*pop_val);
+        *pop_val = NULL;
+    }
+}
+
+void     lst_remove(t_list* list, void* item)
+/*
+        removes an element by value
+
+        Parameters
+        ----------
+        list : t_list*
+            the t_list from where to remove the element
+
+        index : long int
+            the element to be removed
+*/
+{
+    int index;
+    t_pop_ret *ret;
+
+    if ((index = lst_index(list, item)) != -1)
+    {
+        ret = pop(list, (size_t)index);
+        free_pop(&ret);
+    }
+}
+
+void     reverse(t_list* list)
+/*
+        reverse the order of elements in a t_list
+
+        Parameters
+        ----------
+        list : t_list*
+            the t_list to be reversed
+*/
+{
+    size_t tsize = tSizes[list->type];
+    size_t index = 0;
+    char *elements = (char*)(list->elements);
+    void *buffer = malloc(tsize);
+
+    while (index + 1 <= list->size / 2)
+    {
+        memcpy(buffer, &elements[index * tsize], tsize);
+        memcpy(&elements[index * tsize], &elements[(list->size - index - 1) * tsize], tsize);
+        memcpy(&elements[(list->size - index - 1) * tsize], buffer, tsize);
+        index++;
+    }
+    free(buffer);
+}
+
+
+void*   array_subscripting(t_list *list, size_t index)
+/*
+        provides access to a specific element in a t_list
+
+        Parameters
+        ----------
+        list : t_list*
+            the t_list of the element to be accessed
+
+        index : long int
+            the element to be accessed
+*/
+{
+    char *elements = (char*)(list->elements);
+
+    if (list->size > index)
+        return ((void*)(&elements[index * tSizes[list->type]]));
+    return (NULL);
+}
+
+// Sorting ////////////////////////////
+
+/**
+ * @brief Set the length and type object for group allocation.
+ * 
+ * @param list 
+ * @param len 
+ * @param type 
+ */
+void set_length_and_type(t_list *list, size_t *len, int8_t *type)
+{
+    t_list *tmp;
+
+    tmp = list;
+    *len = tmp->size;
+    while (tmp->type == lst_list)
+    {
+        tmp = ((t_list **)(tmp->elements))[0];
+        *len *= tmp->size;
+    }
+    *type = tmp->type;
+}
+
+/**
+ * @brief Branch recursively through every element in the list and copy 
+ * it's value in the array pointed to by `int8_t *group`.
+ * Returns the number of elements copied on every recursion.
+ * 
+ * @param list 
+ * @param group 
+ * @param offset 
+ * @return size_t 
+ */
+size_t collect_data(t_list *list, int8_t *group, size_t offset)
```

Comment:
Why `int8_t *`? Is this equivalent to the `char*` used elsewhere for addresses? I think it would be better to be consistent. Personally I prefer `void*` for addresses as it leads to less type confusion, but I don't know if this can cause problems in certain conditions?

Reply:
This function is removed from the code.